### PR TITLE
Build: Switch from JSHint/JSCS to ESLint (+ other housekeeping)

### LINKED ISSUE #236: Switch to ESLint
We should switch from JSHint+JSCS to ESLint as jQuery Core has.

### PR DESCRIPTION
**Pull request by mgol**
Fixes #236

It makes sense to review it commit by commit, the changes make sense to land separately.

### CODE REVIEW

**Comment by mgol** [comment]
ping @dmethvin. :) @timmywil gave an LGTM.

File: package.json [hunk 1]
```diff
@@ -1,80 +1,81 @@
 {
-	"name": "jquery-migrate",
```
Comment (mgol) [inline package.json]: Why? It already says "2 spaces" so this file was incorrectly formatted...

Comment (markelog) [inline package.json]: O, can't really see tabs vs spaces thingy in github UI

File: Gruntfile.js [hunk 2]
```diff
@@ -136,23 +118,15 @@ module.exports = function( grunt ) {
...
+	require( "load-grunt-tasks" )( grunt );
 
 	// Integrate jQuery migrate specific tasks
 	grunt.loadTasks( "build/tasks" );
 
 	// Just an alias
 	grunt.registerTask( "test", [ "qunit" ] );
 
-	grunt.registerTask( "lint", [ "jshint", "jscs" ] );
+	grunt.registerTask( "lint", [ "eslint:dev", "eslint:dist" ] );
```
Comment (markelog) [inline Gruntfile.js]: Why two tasks? Why not just `eslint`?

File: src/migrate.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
+/* exported migrateWarn, migrateWarnFunc, migrateWarnProp */
```
Comment (markelog) [inline src/migrate.js]: What this means?

File: Gruntfile.js (hunk 2 above)
Comment (mgol) [inline Gruntfile.js]: This is one task with 2 targets. Not specifying the targets manually would first invoke the `dist` one and then quit; that makes it way harder to know what to fix.

Comment (markelog) [inline Gruntfile.js]: > Not specifying the targets manually would first invoke the dist one and then quit

How this is an issue?

> that makes it way harder to know what to fix

Didn't it prints path to problematic file?

**Comment by mgol** [comment]
The problematic file in question would always be dist/jquery-migrate.js.
Hardly a good experience! I want this file to be pointed at only if we
already verified that source files pass the linter.

**Comment by markelog** [comment]
@mgol Make sense

**Comment by markelog** [comment]
Don't be so defensive btw :)

**Comment by mgol** [comment]
I wasn't trying to be defensive, sorry if it sounded that way. :)

**Pull request merged**